sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Helpers/VisualBasicInvocationTracker.cs

```diff
@@ -43,7 +43,32 @@ public VisualBasicInvocationTracker(IAnalyzerConfiguration analysisConfiguration
         protected override SyntaxNode GetIdentifier(SyntaxNode invocationExpression) =>
             ((InvocationExpressionSyntax)invocationExpression).Expression.GetIdentifier();
 
-        protected override Func<MethodSignature, bool> IsSame<TSymbolType>(SyntaxNode identifier, SemanticModel semanticModel) =>
-            MethodSignatureHelper.IsSame<TSymbolType>((SimpleNameSyntax)identifier, semanticModel);
+
+        #region Syntax-level checking methods
+
+        public override InvocationCondition MatchSimpleNames(params MethodSignature[] methods)
+        {
+            return Check;
+
+            bool Check(InvocationContext invocationContext)
```

Comment:
you could directly `return (invocationContext) => { ... }`, I don't think this will reduce the readability a lot...